src/hooks/useLoginProfile.ts

```diff
@@ -0,0 +1,183 @@
+import {
+  CodeChallengeMethod,
+  exchangeCodeAsync,
+  makeRedirectUri,
+  refreshAsync,
+  ResponseType,
+  useAuthRequest
+} from 'expo-auth-session';
+import { useEffect, useState } from 'react';
+
+import * as appJson from '../../app.json';
+import { secrets } from '../config';
+import { addToStore, readFromStore } from '../helpers';
+
+const PROFILE_ACCESS_TOKEN = 'profileAccessToken';
+
+const namespace = appJson.expo.slug as keyof typeof secrets;
+const profile = secrets[namespace]?.profile;
+const serverUrl = profile?.serverUrl;
+
+const keycloakDiscovery = {
+  authorizationEndpoint: serverUrl + '/auth',
+  tokenEndpoint: serverUrl + '/token',
+  revocationEndpoint: serverUrl + '/revoke'
+};
+
+const CLIENT_ID = profile?.clientId;
+const CLIENT_SECRET = profile?.clientSecret;
```

Comment:
what about using these values from global settings to be able to change them on the run instead of having to make new releases as they are shipped in the secrets file?

Reply:
Yeah, you're right. It can be done. We've done it before. But I have some fears about it. Like security. After all, I need clientId and secretKey here and if we get them from `globalSettings`, there may be a security problem, right? After all, this request comes to the app as a result of a request and these requests can be easily intercepted nowadays.